src/main.rs

```diff
@@ -25,125 +26,145 @@ fn update_diagrams_recursively(dir: &Path) -> Result<(), Box<dyn Error>> {
     Ok(())
 }
 
+#[derive(Parser)]
+#[clap(name = "ts_depslint")]
 /// ts_deplint is a tool for linting TypeScript projects for disallowed imports.
-///
-/// Usage:
-///
-///    ts_deplint <command> <path1> <path2> ...
-///
-/// Commands:
-///
-///    lint     Lint the passed-in paths for disallowed imports.
-///    diagram  Update the README.md file in the passed-in paths with a diagram of the disallowed imports.
-///    fix      Fix the disallowed imports in the passed-in paths by adding allow rules.
-///    format   Format the rules files in the passed-in paths.
-///
-/// Examples:
-///
-///    ts_deplint lint src
-///    ts_deplint lint src/domain src/app
-///    ts_deplint lint src/domain/user.ts
-///    ts_deplint diagram src/.deplint.rules.yml
-///    ts_deplint diagram src
-///    ts_deplint fix src
-///    ts_depllint format src/.deplint.rules.yml
-///    ts_depllint format src
-///
-/// Paths:
-///
-///  Paths can be either directories or files.
-///
-/// # Errors
-///
-/// This function will return an error if anything goes wrong.
+struct Opt {
+    #[clap(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    Lint(LintCommand),
+    Diagram(DiagramCommand),
+    Fix(FixCommand),
+    Format(FormatCommand),
+}
+
+#[derive(Parser)]
+#[clap(rename_all = "camel_case")]
+/// Lint the passed-in paths for disallowed imports.
+struct LintCommand {
+    /// Paths can be either directories or files.
+    paths: Vec<String>,
+}
+
+#[derive(Parser)]
+#[clap(rename_all = "camel_case")]
+/// Update the README.md file in the passed-in paths with a diagram of the disallowed imports.
+struct DiagramCommand {
+    /// Paths can be either directories or files.
+    paths: Vec<String>,
+}
+
+#[derive(Parser)]
+#[clap(rename_all = "camel_case")]
+/// Fix the disallowed imports in the passed-in paths by adding allow rules.
+struct FixCommand {
```

Comment:
I omitted `#[arg(last = true)]` to avoid the for `--`:

https://github.com/maxh/ts_deplint/pull/1/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR49

https://docs.rs/clap/latest/clap/struct.Arg.html#method.last